app/controllers/contact.js

```diff
@@ -1,31 +1,19 @@
 import Ember from 'ember';
 import sharedActions from '../mixins/shared-actions';
 
+import MaintainerValidator from '../validator/maintainer';
 
 export default Ember.Controller.extend(sharedActions, {
-	name: null,
-	email: null,
-	emailMissing: false,
-	emailError: Ember.computed('emailMissing', 'email', function(){
-		if (this.get('emailMissing') === true && this.get('email') === null){
-			return true;
-		} else {
-			return false;
-		}
-	}),
-
+	MaintainerValidator,
+	showErrorState: false,
 	actions: {
-		changeRoute: function(route){
-			if (this.get('email')) {
-				this.store.peekAll('submission').get('firstObject').set('maintainer_email', this.get('email'));
-				if (this.get('name')){
-					this.store.peekAll('submission').get('firstObject').set('maintainer_name', this.get('name'));
-				}
-				this.transitionToRoute(route);
-			} else {
-				(this.set('emailMissing', true));
-			}
+		showError: function () {
```

Comment:
Maybe we could add some logic here to only show the error when the `next` button has been clicked, after the user has already clicked and has re-entered a field to correct it.

Reply:
We talked about this real life. Let's wait how `ember-changeset-validatations` handles this until the end of month. (Because current option that changeset offers to turn off automatic validation doesn't work with changeset-valdation) If nothing on the package side, we will do something about it.